checks/file_utils.go

```diff
@@ -131,6 +131,61 @@ func CheckFilesContent(shellPathFnPattern string,
 	return res, nil
 }
 
+// FileCbData is any data the caller can act upon
+// to keep state.
+type FileCbData interface{}
+
+// FileContentCb is the callback.
+// The bool returned indicates whether the CheckFilesContent2
+// should continue iterating over files or not.
+type FileContentCb func(path string, content []byte,
+	dl checker.DetailLogger, data FileCbData) (bool, error)
+
+func CheckFilesContent2(shellPathFnPattern string,
```

Comment:
note: once this PR is merged, I will rename it to `CheckFilesContent` and update other uses of it. Same of `CheckForFile`.